### DISCUSSION TITLE
Channel 有不阻塞的方式获取元素吗?

### DISCUSSION THREAD

**Discussion by ShirenY** [idealvin/coost#375] [Q&A] [answered]
写的那端阻塞我觉得没问题. 但是读的这端在读取的同时还需要做别的事情, 一直阻塞就卡住了. golang中可以用select, coost好像没找到相关接口. 难道是设置0 timeout?

**Comment by idealvin** [comment, marked as answer]
理论上是这样。不过channel性能很差，你自己随便用个队列，加锁，都比channel强

**Reply by ShirenY** [reply]
太实诚了。为什么性能差能解释一下吗？channel不也可以靠锁和队列实现？
